source/plugins/go/input/lib/applicationinsights.go

```diff
@@ -0,0 +1,306 @@
+	}
+
+	aiLogger = CreateLogger(logPath)
+	// Initialize customProperties
+	customProperties = make(map[string]string)
+	if !isIgnoreProxySettings() {
+		if isWindows {
+			proxyEndpoint = os.Getenv("PROXY")
+
+		} else {
+			proxyEndpoint = getProxyEndpoint()
+		}
+	}
+
+	// Retrieve environment variables
+	resourceInfo := os.Getenv("AKS_RESOURCE_ID")
+	if resourceInfo == "" {
+		customProperties["ID"] = os.Getenv(envAcsResource)
+		customProperties["Region"] = ""
+	} else {
+		customProperties["ID"] = resourceInfo
+		customProperties["Region"] = os.Getenv(envAksRegion)
+	}
+
+	customProperties["WSID"] = getWorkspaceId()
+	customProperties["Version"] = os.Getenv(envAgentVersion)
+	customProperties["Controller"] = controllerType[strings.ToLower(os.Getenv(envController))]
+	customProperties["Computer"] = hostName
+	encodedAppInsightsKey := os.Getenv(envAppInsights)
+	appInsightsEndpoint := os.Getenv(envAppEndpoint)
+	customProperties["WSCloud"] = getWorkspaceCloud()
+	isProxyConfigured := false
+	if proxyEndpoint != "" {
+		customProperties["Proxy"] = "true"
+		isProxyConfigured = true
+		if isProxyCACertConfigured() {
+			customProperties["IsProxyCACertConfigured"] = "true"
+		}
+	} else {
+		customProperties["Proxy"] = "false"
+		isProxyConfigured = false
+		if isIgnoreProxySettings() {
+			customProperties["IsProxyConfigurationIgnored"] = "true"
+		}
+	}
+
+	isMSI := os.Getenv(envAADMSIAuth)
+	if isMSI != "" && strings.EqualFold(isMSI, "true") {
+		customProperties["isMSI"] = "true"
+	} else {
+		customProperties["isMSI"] = "false"
+	}
+
+	addonResizerVPAEnabled := os.Getenv(envAddonResizer)
+	if addonResizerVPAEnabled != "" && strings.EqualFold(addonResizerVPAEnabled, "true") {
+		customProperties["addonResizerVPAEnabled"] = "true"
+	}
+
+	if encodedAppInsightsKey != "" {
+		decodedAppInsightsKey, err := base64.StdEncoding.DecodeString(encodedAppInsightsKey)
+		if err != nil {
+			aiLogger.Printf("Error decoding Application Insights key: %s", err.Error())
+		}
+
+		telemetryClientConfig := appinsights.NewTelemetryConfiguration(string(decodedAppInsightsKey))
+
+		if appInsightsEndpoint != "" {
+			aiLogger.Printf("Setting Application Insights endpoint to %s", appInsightsEndpoint)
+			telemetryClientConfig.EndpointUrl = appInsightsEndpoint
+		}
+
+		if isProxyConfigured {
+			proxyEndpointUrl, err := url.Parse(proxyEndpoint)
+			if err != nil {
+				aiLogger.Printf("Error parsing proxy endpoint: %s", err.Error())
+				telemetryClient = nil
+				return
+			}
+
+			transport := &http.Transport{
+				Proxy: http.ProxyURL(proxyEndpointUrl),
+			}
+			httpClient := &http.Client{
+				Transport: transport,
+			}
+			telemetryClientConfig.Client = httpClient
+		}
+		telemetryClient = appinsights.NewTelemetryClientFromConfig(telemetryClientConfig)
+	}
+
+	telemetryOffStr := os.Getenv("DISABLE_TELEMETRY")
+	if strings.EqualFold(telemetryOffStr, "true") {
+		telemetryClient.SetIsEnabled(false)
+	}
+
+}
+
+func getContainerRuntimeInfo() {
```

Comment:
looks like we dont need function for this and instead we just read the environment variable in the initialization and use that every where in place. thoughts?